Say Their Names/Source/Model/Person.swift

```diff
@@ -7,23 +7,30 @@
 //
 
 import Foundation
+import FirebaseFirestore
+import FirebaseFirestoreSwift
 
 struct Person: Codable {
-    var id: String
-    var fullName: String
-    var age: Int
-    var childrenCount: Int
-    var date: Date
-    var location: String
-    var media: [String]
-    var bio: String
-    var context: String
-    var donations: [Donation]
-    var petitions: [Petition]
+    let name: String
+    let age: Int
+    let numberOfChildren: Int
+    
+    // TODO: ensure what timezone dates are in
+    let dateOfIncident: Date
+    let dateOfBirth: Date
+
+    let location: DocumentReference
+    
+//    let media: [String]
+    let picture: String
+    let bio: String
+    let context: String
+    let donations: [DocumentReference]
+    let petitions: [DocumentReference]
```

Comment:
Haven't tried this approach before, but wouldn't it be better to work with first party objects as opposed to `DocumentReference`s ?

Reply:
As far as I understand there is no “deep fetch” feature in Firestore. So we only get references and not full objects, needing to fetch them separately. 
I would prefer to have fetched objects directly too!
One option is to duplicate the data, but I was hesitant to start with that optimization right away.
What do you think?

Reply:
We should use https://github.com/alickbass/CodableFirebase.

Reply:
We can try that as a next step from here 👍 so far I followed official documentation

Reply:
IMO our first party objects should resemble the document's structure; meaning that we should only keep an array of `donationIds` and `petitionIds` if that is how we plan on structuring the data